std/path.d

```diff
@@ -3956,23 +3956,31 @@ string expandTilde(string inputPath) nothrow
         }
 
         // Replaces the tilde from path with the environment variable HOME.
-        static string expandFromEnvironment(string path) nothrow
+        static string expandFromEnvironment(string path) nothrow @safe
         {
             import core.stdc.stdlib : getenv;
+            import core.stdc.string : strlen;
+            import std.exception : assumeUnique;
 
             assert(path.length >= 1);
             assert(path[0] == '~');
 
             // Get HOME and use that to replace the tilde.
-            auto home = getenv("HOME");
-            if (home == null)
-                return path;
+            auto home = () @trusted {
+                auto homeStr = getenv("HOME");
+                if (homeStr == null)
+                    return path;
+                return homeStr[0 .. strlen(homeStr)].assumeUnique;
```

Comment:
`assumeUnique` can be done outside of the @trusted lambda.

Reply:
`assumeUnique` is `@system`